modules/markup/html.go

```diff
@@ -657,6 +658,14 @@ func sha1CurrentPatternProcessor(ctx *postProcessCtx, node *html.Node) {
 	// but that is not always the case.
 	// Although unlikely, deadbeef and 1234567 are valid short forms of SHA1 hash
 	// as used by git and github for linking and thus we have to do similar.
+	// Because of this, we check to make sure that a matched hash is actually
+	// a commit in the repository before making it a link.
+	if ctx.metas["repoPath"] != "" {
+		if _, err := git.NewCommand("log", "-1", hash).RunInDirBytes(ctx.metas["repoPath"]); err != nil {
```

Comment:
I think it would be better to use src-d/git to check this instead of running git command as it is quite a performance penalty

Reply:
Done

Reply:
So the correct call for git would be `git.NewCommand("rev-parse", "--verify", reference).RunInDir(repo.RepoPath())`

`git log` is quite slow. 

Unfortunately as far as I can see src.d/go-git hasn't really implemented an equivalent of rev-parse. https://github.com/src-d/go-git/issues/599

Reply:
From another PR:

https://github.com/go-gitea/gitea/blob/a500d83769c44826e89a5a6fe87434779c430262/modules/git/repo_commit.go#L120-L135

Reply:
> So the correct call for git would be `git.NewCommand("rev-parse", "--verify", reference).RunInDir(repo.RepoPath())`
> 
> `git log` is quite slow.
> 
> Unfortunately as far as I can see src.d/go-git hasn't really implemented an equivalent of rev-parse. [src-d/go-git#599](https://github.com/src-d/go-git/issues/599)

I'll take a look at this. I didn't test with short hashes so didn't realize go-git didn't support them.